### PR TITLE
chore: code and configs cleanups; add builtin cache tracing

### PR DESCRIPTION
**Pull request by PGNetHun**
### Description of the feature or fix

1. Clean up some code and config files: fix indentations, macros, and naming conventions
2. Add builtin cache manager tracing

### Checkpoints
- [ ] Run `code-format.py` from the scripts folder. [astyle](http://astyle.sourceforge.net/install.html) needs to be installed.
- [ ] Update the [Documentation](https://github.com/lvgl/lvgl/tree/master/docs) if needed
- [ ] Add [Examples](https://github.com/lvgl/lvgl/tree/master/examples) if relevant. 
- [ ] Add [Tests](https://github.com/lvgl/lvgl/blob/master/tests/README.md) if applicable.
- [ ] If you added new options to `lv_conf_template.h` run [lv_conf_internal_gen.py](https://github.com/lvgl/lvgl/blob/release/v8.3/scripts/lv_conf_internal_gen.py) and update [Kconfig](https://github.com/lvgl/lvgl/blob/release/v8.3/Kconfig).
 
Be sure the following conventions are followed:
- [ ] Follow the [Styling guide](https://github.com/lvgl/lvgl/blob/master/docs/CODING_STYLE.md)
- [ ] Prefer `enum`s instead of macros. If inevitable to use `define`s export them with `LV_EXPORT_CONST_INT(defined_value)` right after the `define`.
- [ ] In function arguments prefer `type name[]` declaration for array parameters instead of `type * name`
- [ ] Use typed pointers instead of `void *` pointers
- [ ] Do not `malloc` into a static or global variables. Instead declare the variable in `lv_global_t` structure in [`lv_global.h`](https://github.com/lvgl/lvgl/blob/master/src/core/lv_global.h) and mark the variable with `(LV_GLOBAL_DEFAULT()->variable)` when it's used. See a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#memory-management).
- [ ] Widget constructor must follow the `lv_<widget_name>_create(lv_obj_t * parent)` pattern.
- [ ] Widget members function must start with `lv_<modul_name>` and should receive `lv_obj_t *` as first argument which is a pointer to widget object itself.  
- [ ] `struct`s should be used via an API and not modified directly via their elements.
- [ ] `struct` APIs should follow the widgets' conventions. That is to receive a pointer to the `struct` as the first argument, and the prefix of the `struct` name should be used as the prefix of the function name too (e.g.  `lv_disp_set_default(lv_disp_t * disp)`)
- [ ] Functions and `struct`s which are not part of the public API must begin with underscore in order to mark them as "private".
- [ ] Arguments must be named in H files too.
- [ ] To register and use callbacks one of the following needs to be followed (see a detailed description [here](https://docs.lvgl.io/master/get-started/bindings/micropython.html#callbacks)):
  - For both the registration function and the callback pass a pointer to a `struct` as the first argument. The `struct` must contain `void * user_data` field.
  - The last argument of the registration function must be `void * user_data` and the same `user_data` needs to be passed as the last argument of the callback.
  - Callback types not following these conventions should end with `xcb_t`.

### CODE REVIEW

File: src/draw/lv_image_decoder.c [hunk 1]
```diff
@@ -94,10 +94,10 @@ lv_result_t lv_image_decoder_get_info(const void * src, lv_image_header_t * head
     }
 
     lv_result_t res = LV_RESULT_INVALID;
-    lv_image_decoder_t * d;
-    _LV_LL_READ(img_decoder_ll_p, d) {
-        if(d->info_cb) {
-            res = d->info_cb(d, src, header);
+    lv_image_decoder_t * decoder;
```
Comment (kisvegabor) [inline src/draw/lv_image_decoder.c:97]: Is there a particular reason for this change? I'm ok with that, just want to be sure if it's really just a minor refactoring.

File: src/misc/lv_anim.c [hunk 2]
```diff
@@ -45,9 +45,9 @@ static int32_t lv_anim_path_cubic_bezier(const lv_anim_t * a, int32_t x1,
  *      MACROS
  **********************/
 #if LV_LOG_TRACE_ANIM
-    #define TRACE_ANIM(...) LV_LOG_TRACE(__VA_ARGS__)
+    #define LV_TRACE_ANIM(...) LV_LOG_TRACE(__VA_ARGS__)
 #else
-    #define TRACE_ANIM(...)
+    #define LV_TRACE_ANIM(...)
```
Comment (kisvegabor) [inline src/misc/lv_anim.c:50]: It's not really a global policy but rather my personal preference to add the `lv_` prefix only to global functions/enums/types to differentiate them from static and local declarations. 

So do you thin kit's easier to read the code this way?

**Review by kisvegabor — COMMENTED** [review]
All looks good, thank you! 
Just a few simple questions.

File: src/draw/lv_image_decoder.c (hunk 1 above)
Comment (PGNetHun) [reply src/draw/lv_image_decoder.c:97]: No specific reason, I just like to follow naming conventions, and clean code... meaning: avoid short (1-2 letters) names and use descriptive (full length) names.

File: src/misc/lv_anim.c (hunk 2 above)
Comment (PGNetHun) [reply src/misc/lv_anim.c:50]: The aim is to have same naming convention in C files where we use tracing macros.
I saw that the tracing macro name differs across files, so I thought to follow the naming convention: LV_TRACE_xxxx.

**Comment by PGNetHun** [comment]
PS: there will be another PRs cleaning up code and documentation, because during MicroPython testing (https://github.com/lvgl/lv_binding_micropython/pull/292) I've found some inconsistency in code and documentation.

File: src/misc/lv_anim.c (hunk 2 above)
Comment (PGNetHun) [reply src/misc/lv_anim.c:50]: Could you please advise me the name convention to this tracing macro?
I will amend the PR with the new names.

Comment (kisvegabor) [reply src/misc/lv_anim.c:50]: Oh, I haven't noticed that it's used inconsistently. Let's use your suggestion :+1:

File: src/draw/lv_image_decoder.c (hunk 1 above)
Comment (kisvegabor) [reply src/draw/lv_image_decoder.c:97]: Personally I'm ok with one letter names for short scope. However, I'm in favor of letting people leaving their footprint in LVGL, so I would be happy to keep these changes.

**Comment by kisvegabor** [comment]
All clear, I merge this one! :rocket: 

> there will be another PRs cleaning up code and documentation, because during MicroPython testing (https://github.com/lvgl/lv_binding_micropython/pull/292)

I'm looking into it right now!

> As these cleanup changes are small ones (no code logic change), I can help in making these doc changes.

That would be amazing. We really have backlog with updating the docs.

**Pull request merged**